bin/extract_metrics/extract_metrics.py

```diff
@@ -0,0 +1,339 @@
+"""
+Check all jobs run with versions >= 1.0.0 and < 1.0.8 for metrics outputs in S3.
+If not found, use existing p3d output and analysis params to create and upload metrics output.
+"""
+
+import asyncio
+from dataclasses import asdict
+import datetime
+import json
+import logging
+import logging.config
+import os
+import tempfile
+from typing import Any
+from zipfile import ZipFile
+
+import asyncpg
+import boto3
+import polars as pl
+from pulse3D import metrics
+from pulse3D.data_loader.utils import get_metadata_cls
+from pulse3D.data_loader import InstrumentTypes
+from pulse3D.peak_finding import LoadedDataWithFeatures
+from pulse3D.data_loader.metadata import NormalizationMethods
+from pulse3D.pre_analysis import sort_wells_in_df, apply_window_to_df, PreAnalyzedData, post_process
+import structlog
+from utils.s3 import upload_file_to_s3
+
+PULSE3D_UPLOADS_BUCKET = os.getenv("TEST_UPLOADS_BUCKET")
+
+DRY_RUN = True
+
+s3_client = boto3.client("s3")
+
+logging.config.dictConfig(
```

Comment:
why the two different loggers?  does it merit a comment here stating the reason?

Reply:
p3d uses structlog which unfortunately means we have to go through this complicated setup to get logs written to both stdout and a file. I can add a comment with links explaining this